### PR TITLE
Local storage

### PR DESCRIPTION
**Pull request by Mutai-Gilbert**
In this pull request I made the following changes:
-  I used local storage to save user input.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -285,20 +285,52 @@ window.onload = () => {
...
+  };
+
+  const buttonReset = document.getElementById('reset');
+  buttonReset.addEventListener('click', reset);
 }
 
 const form = document.getElementById('form');
-form.addEventListener('submit', onSubmit);
\ No newline at end of file
+form.addEventListener('submit', onSubmit);
```
Comment (pinkmoon25) [inline app.js:340]: - [x] Great job on implementing the localstorage :clap: , However we need to store the form data as soon as it is filled, currently you save the data only when the user submits the form, you an use the change event listener to save data to local storage once the form fields are filled.
- [x] Also Please make sure when there is any form data in the local storage the form is populated with that data on page reload, you can use the reload event listener and check the local storage and if there is any data available populate your form with the appropriate data.
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```
The points to consider : 
1. The contact form information is persistent to page reloads.
2. After filling the form, and not submitting it, the entered values should be saved in the browser.
3. When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.
4. If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty.

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @Mutai-Gilbert @lilskyex0x , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
Linters checks :heavy_check_mark: 
Prefessional ReadMe :heavy_check_mark: 
Pair-programing :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
[OPTIONAL] I think it would be better if you use a separate js file for this functionality like localStoragre.js

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js [hunk 2]
```diff
@@ -284,21 +284,42 @@ window.onload = () => {
...
+  };
+  localStorage.setItem('data', JSON.stringify(Data));
+});
 
-const form = document.getElementById('form');
-form.addEventListener('submit', onSubmit);
\ No newline at end of file
+const AutoFiller = localStorage.getItem('data');
+if (AutoFiller) {
+  const DataStored = JSON.parse(localStorage.getItem('data'));
```
Comment (rloterh) [inline app.js]: - [x] Filled data in the form input are not reflecting in the local storage. Whiles a user is entering a data (_onChange event_), those data need to reflect in the local storage on the fly, as mentioned in the previous review. As you can see from the screenshot below, the entered data in the form are not being persisted to the local storage. Kindly ensure to improve upon the same.
- [x] Also, entered data in the form inputs needs to remain in the input fields when the page is refreshed

image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Mutai & Lilsnow, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good job with implementations so far!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- Kindly include details of the co contributor(s) in your ReadMe.
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js [hunk 3]
```diff
@@ -284,21 +284,69 @@ window.onload = () => {
...
+const formComment = document.getElementById('textarea');
+
+// Store formData in localStorage
+function loadLocalStorage() {
+  const formData = JSON.parse(window.localStorage.getItem('formData'));
+  if (formData) {
+    formName.value = formData.name;
+    formEmail.value = formData.email;
+    formComment.value = formData.message;
+  }
```
Comment (EricMbouwe) [inline app.js]: - [ ] There is a bug here making the `Textarea` value `undefined` when the page is reloaded. `formData.message` does not exist, according to your formData object, it should be formData.textarea. Let's fix that
Screenshot 2022-11-19 at 1 10 18 PM

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @Mutai-Gilbert, 

Great work on making the changes requested by a previous reviewer 👏 🔥
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights 🎉🔥
- Fetch from localStorage ✔
- No linter errors ✔

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Santiago220991 — APPROVED** [review]
Hi @Mutai-Gilbert, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
:heavy_check_mark: Applied changes suggested by the last reviewer.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Followed GitHub flow.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**